transport/split/stream_dialer.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2023 Jigsaw Operations LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package split
+
+import (
+	"context"
+	"errors"
+	"io"
+
+	"github.com/Jigsaw-Code/outline-sdk/transport"
+)
+
+type splitWriter struct {
+	writer      io.Writer
+	splitPoint  int64
+	bytesCopied int64
+}
+
+var _ io.ReaderFrom = (*splitWriter)(nil)
+
+func (w *splitWriter) ReadFrom(source io.Reader) (int64, error) {
+	var written int64
+	defer func() {
+		w.bytesCopied += written
+	}()
+	n, err := io.CopyN(w.writer, source, w.splitPoint-w.bytesCopied)
```

Comment:
What is the purpose of splitting the traffic here?

Reply:
Because some bytes may had been copied already from previous writes, so you need to discount them to hit the splitPoint correctly.